packages/config/src/class-config.php

```diff
@@ -71,6 +71,38 @@ public function ensure( $feature, array $options = array() ) {
 		$this->set_feature_options( $feature, $options );
 	}
 
+	/**
+	 * Require multiple features to be initialized. It's up to the package consumer to actually add
+	 * the package to their composer project. Declaring a requirement using this method
+	 * instructs the class to initialize it.
+	 *
+	 * Usage example:
+	 * <code>
+	 * $config->ensure_bunch(
+	 *     array(
+	 *         'slug'        => 'an-example-plugin',
+	 *         'name'        => 'An Example Plugin,
+	 *         'url_info'    => 'https://example.org',
+	 *         'plugin_file' => __FILE__,
+	 *     ),
+	 *     array(
+	 *         'connection' => array(
+	 *             'deactivate_disconnect' => true,
+	 *             'tos_link'              => true,
+	 *         )
+	 *     )
+	 * );
+	 * </code>
+	 *
+	 * @param array $common_options The common options to be passed into each feature's initialization method.
+	 * @param array $features       Features to be enabled and their options.
+	 */
+	public function ensure_bunch( array $common_options, array $features ) {
```

Comment:
I know this is still in progress, but I'm excited! 

Do we need a new method? Can we not refactor the ->ensure() method to accept multiple features?

Reply:
Yes, we can modify `ensure()` and check if the first argument is an array or a string.
I think it would make the code more complex, and having a separate passthrough method is easier to understand for the package users.

It's syntax is already complex (see the phpdoc example), and trying to merge it into the existing `ensure()` argument format will make it even even more complicated and error-prone.

Reply:
I guess it's a question of what is more important to us: Code complexity or usability/maintenance complexity? 

I feel like it's not ideal to have multiple methods that can `ensure` a feature/package that we need to maintain. This means double the tests, double to back-compat considerations, and double the entry points and syntaxes for setting up a plugin that we need to consider. 

I'm open to discuss further though, wonder what the team thinks.

Reply:
> This means double the tests

We will still need separate tests for `ensure()` to check its behavior in the regular and multi-option mode.
And since `ensure_bunch()` is generally a passthrough method, we won't need to duplicate all the same tests we write for `ensure()`.
There are no tests for the `Config` package at the moment though ¯\\\_(ツ)\_/¯

I'll bring it up with the team to discuss.

Reply:
ok right, to me this is the biggest consideration: `double the entry points`. But worth a discussion!